dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -176,10 +182,49 @@ private void cacheInit() {
+                            tcInvoke(cacheStoreService.tcCacheManager, "createCache", metaCacheName, config);
+                        
+                        // Get RemoteCacheManager cm field from remote JCacheManager
+                        Field cmField = cachingProviderClass
+                                        .getDeclaredField("cm");
+                        cmField.setAccessible(true);
+                        Object remoteCacheManger = cmField.get(cacheStoreService.cacheManager);
+
+                        // Object remoteCacheManagerAdmin = remoteCacheManager.administration();
+                        Object remoteCacheManagerAdmin = remoteCacheManger.getClass()
+                                        .getMethod("administration")
+                                        .invoke(remoteCacheManger);
+
+                        // Create replicated Infinispan configuration XML for this metaCacheName.
+                        String infinispanConfig = String.format("<infinispan>" +
+                                        "<cache-container>" +
+                                        "<replicated-cache name=\"%s\"/>" +
+                                                               "</cache-container>" +
+                                                             "</infinispan>",
+                                                             metaCacheName);
+                        
+                        // XMLStringConfiguration xMlStringConfiguration = new XMLStringConfiguration(infinispanConfig)
+                        Class<?> XMLStringConfigurationClass = cachingProviderClass.getClassLoader().loadClass("org.infinispan.commons.configuration.XMLStringConfiguration");
+                        Object xMlStringConfiguration = XMLStringConfigurationClass.getConstructor(String.class)
+                        .newInstance(infinispanConfig.toString());
+                        
+                        // remoteCacheManagerAdmin.getOrCreateCache(metaCacheName, xMlStringConfiguration);
+                        remoteCacheManagerAdmin.getClass()
+                        .getMethod("getOrCreateCache", String.class, cachingProviderClass.getClassLoader().loadClass("org.infinispan.commons.configuration.BasicConfiguration"))
+                        .invoke(remoteCacheManagerAdmin, metaCacheName, xMlStringConfiguration);
+                        
+                        // Refresh JCacheManager
+                        cacheStoreService.cacheManager = cacheStoreService.cachingProvider.getCacheManager(cacheStoreService.cacheManager.getURI(), cacheStoreService.cacheManager.getClassLoader(), cacheStoreService.cacheManager.getProperties());
+                        sessionMetaCache = cacheStoreService.cacheManager.getCache(metaCacheName);
+                    } else {
+                        if (trace && tc.isDebugEnabled())
+                            tcInvoke(cacheStoreService.tcCacheManager, "createCache", metaCacheName, config);
```

Comment:
minor/optional: maybe move this block of Infinispan-specific stuff to a separate method, so that the flow will be more straightforward for general JCache usage?

Reply:
Suggestion taken.